### PR TITLE
[PLAT-512] Fix risky parsing in mds-audit GET /vehicles endpoint

### PR DESCRIPTION
**Pull request by avatarneil**
## 📚 Purpose
Patches a bug where supplying invalid JSON to the audit GET `/vehicles` endpoint via the `bbox` parameter would result in an unhandled error.

## 👌 Resolves:
PLAT-512

## 📦 Impacts:
See changesets

### CODE REVIEW

**Review by drtyh2o — APPROVED** [review]

File: packages/mds-audit-api/api.ts [hunk 1]
```diff
@@ -676,27 +676,42 @@ function api(app: express.Express): express.Express {
...
+          bbox,
+          provider_id
+        } = {
+          ...parseRequest(req)
+            .single({
+              parser: (val, key) => {
+                try {
+                  return JSON.parse(val)
+                } catch (error) {
+                  throw new ValidationError(`invalid query parameter ${key}`, { [key]: val })
```
Comment (nessur) [inline packages/mds-audit-api/api.ts:692]: In general it's a bad idea to return any user input back to them, this is probably fine, but watch out, this is the basis for most buffer or injection attacks. Less helpful error messages are the only way.

**Review by nessur — COMMENTED** [review]
Please consider not including user input in thrown errors.

File: packages/mds-audit-api/api.ts (hunk 1 above)
Comment (janedotx) [reply packages/mds-audit-api/api.ts:692]: Can we log the error with the key/val here, but change the error message in the exception thrown to something more cryptic perhaps?

Comment (avatarneil) [reply packages/mds-audit-api/api.ts:692]: Considering the clients for this API (and basically all others we operate right now) are at least semi-trusted parties that've auth'd, I'm a bit less worried about that scenario. Generally agreed, though.

**Review by PeterSP — APPROVED** [review]

File: packages/mds-audit-api/api.ts (hunk 1 above)
Comment (PeterSP) [reply packages/mds-audit-api/api.ts:692]: At the end of the day, I think `key` can only be from a list provided in our code here, and so could not be 'evil'.

**Review by nessur — APPROVED** [review]

**Pull request merged**